server/repository/repository.go

```diff
@@ -48,9 +48,14 @@ func NewServer(
 	}
 }
 
-func (s *Server) getConnectionState(ctx context.Context, url string) appsv1.ConnectionState {
-	if connectionState, err := s.cache.GetRepoConnectionState(url); err == nil {
-		return connectionState
+// Get the connection state for a given repository URL by connecting to the
+// repo and evaluate the results. Unless forceRefresh is set to true, the
+// result may be retrieved out of the cache.
+func (s *Server) getConnectionState(ctx context.Context, url string, forceRefresh bool) appsv1.ConnectionState {
+	if !forceRefresh {
```

Comment:
Why do we need force refresh? I'm worried that we're using the wrong key for the correction state.

Reply:
I think it'd be useful when you change repository credentials, you can see whether the change was successful or not instantly. Credentials aren't (can't be) checked for validity. 

The other possibility would be to invalidate the cache on connection state for every repository that matches the credentials' URL prefix, when the credentials change.